src/pineko/kfactor.py

```diff
@@ -112,11 +107,12 @@ def compute_scale_factor(
     float
         full contribution factor
     """
-    max_as = to_construct_order[0] - m
+    max_as = to_construct_order[0]
     alpha_val = alphas.alphasQ2(Q2)
-    alpha_term = 1.0 / pow(alpha_val, max_as - (nec_order[0] - m))
+    alpha_term = 1.0 / pow(alpha_val, max_as - (nec_order[0]))
     k_term = central_k_factor[bin_index] - 1.0
-    if order_exists and (max_as - (nec_order[0] - m)) == 0:
+    # TODO: why do we need a double condition here ???
+    if order_exists and (max_as - (nec_order[0])) == 0:
```

Comment:
@andreab1997 can you remind me why we need both conditions?

Reply:
You need to think at the equation. However the idea is that we always need the -1.0, but for the case in which the order we are constructing already exists (so we are replacing it) and, even in this case, only if nec_order is exactly this order (while nec_orders loop over all the orders in the grid anyway)

Reply:
This is the reason we need both. You clearly see what is happening with a pdb

Reply:
I think I manage to solve it a bit easily. because whenever you will reach the max order you can get $1/as^0$, which in the end is 1. I need to look at it once more though